src/components/Dot/index.tsx

```diff
@@ -0,0 +1,34 @@
+import styled from '@emotion/styled'
+import PropTypes from 'prop-types'
+import React, { FunctionComponent } from 'react'
+import { Color } from '../../theme/colors'
+import Box from '../Box'
+
+const StyledDot = styled(Box, {
+  shouldForwardProp: prop => !['color'].includes(prop.toString()),
+})`
+  display: inline-block;
+  border-radius: 50%;
+  width: 10px;
+  height: 10px;
+  background-color: ${({ theme: { colors }, color }) =>
+    colors[color as Color] ?? color};
+`
+
+type Props = {
+  color?: string
+  onClick?: () => unknown
+} & XStyledProps
+
+const Dot: FunctionComponent<Props> = ({ color = 'primary', ...props }) => (
+  <StyledDot color={color} {...props} />
+)
+
+Dot.propTypes = {
+  /**
+   * The dot color
+   */
```

Comment:
You need to move the comment to type declaration or it won't be displayed on storybook

Reply:
It works for me in the proptypes? Same in other comments. And if I move it to types it doesn't work anymore.

Reply:
After some testing it looks like we leave it like this in proptypes. Comments in types work with react-docgen-typescript but not react-docgen which we are using now to support extensions on types. @matthprost

Reply:
ok change of answer. We go full TS and remove all comments from proptypes. It won't be displayed into Storybook but no hacks in our codebase just for storybook. @matthprost